hls4ml/templates/vivado/nnet_utils/nnet_merge.h

```diff
@@ -108,6 +115,20 @@ void minimum(
     }
 }
 
+template<class input1_T, class input2_T, class res_T, typename CONFIG_T>
+void dot1d(
+    input1_T data1[CONFIG_T::n_elem], 
+	input2_T data2[CONFIG_T::n_elem],
+    res_T res[CONFIG_T::n_elem_out])
+{
+    res[0] = 0;
+    for (int ii=0; ii<CONFIG_T::n_elem; ii++) {
+      res[0] += data1[ii] * data2[ii];
```

Comment:
You should call `product` function here and store the result in intermediate array and accumulate the result later. Also, add limiter for the number of DSPs allocated. Look into `nnet_dense_latency.h` to see how to do those two things. Also, this function should be pipelined, so add a pragma before the loop.